packages/core/src/input/input-file-resolver.ts

```diff
@@ -117,6 +119,14 @@ export class InputFileResolver {
   }
 
   private async expandPattern(globbingExpression: string, logAboutUselessPatterns: boolean): Promise<string[]> {
+    if (/(:\d+){4}$/.exec(globbingExpression)) {
+      if (hasMagic(globbingExpression)) {
+        throw new StrykerError('Do not use glob patterns with specific mutants on the glob pattern');
```

Comment:
As mentioned, this could be part of the `OptionsValidator`. Throwing a `ConfigError` is better than a `StrykerError`, because it will show a better readable error message to the end-user, without a stack trace.